apm-sniffer/apm-toolkit-activation/apm-toolkit-log4j-1.x-activation/src/main/java/org/apache/skywalking/apm/toolkit/activation/log/log4j/v1/x/log/GRPCLogAppenderActivation.java

```diff
@@ -53,6 +58,10 @@ protected ClassMatch enhanceClass() {
 
     @Override
     public InstanceMethodsInterceptPoint[] getInstanceMethodsInterceptPoints() {
+        if (!ToolkitConfig.Plugin.Toolkit.Log.TRANSMIT_PREFORMATTED) {
+            LOGGER.warn("Log4j 1.x does not support transmitting non-preformatted messages");
```

Comment:
If this is a warning, I think you should return an empty list?

Reply:
Or you should be explicit that, the log still will be in formatted? Is `preformatted` a typo?

Reply:
It's a warning, yes. Returning an empty list would completely disable the ability to transmit any logs whenever they set `transmit_preformatted=true`. Instead, I thought it would be better just to warn that they won't be un-formatted and then continue to send un-formatted.

Preformatted isn't a typo. Just pre + formatted.

Reply:
Updated terminology to formatted & un-formatted.